src/MongoDB.Driver.Legacy/MongoCollection.cs

```diff
@@ -1714,13 +1716,19 @@ private MapReduceResult MapReduce(IClientSessionHandle session, MapReduceArgs ar
                     Collation = args.Collation,
                     Filter = query,
                     FinalizeFunction = args.FinalizeFunction,
+#pragma warning disable 618
```

Comment:
Maybe it's possible to disable these warning once in the scope (with including the fields which are not deprecated). But I think this way is also should work. 
NOTE: I won't mind changing these lines.

Reply:
It's tempting to disable these warnings for the whole method but then in the future if a new field is marked obsolete we might miss a compiler warning if warnings are disabled for the whole method.

Reply:
agreed